pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -104,49 +188,19 @@ func (pl *VolumeZone) Filter(ctx context.Context, _ *framework.CycleState, pod *
 		return nil
 	}
 
+	state, err := getStateData(cs)
+	if err != nil {
+		return framework.AsStatus(err)
+	}
+
 	for i := range pod.Spec.Volumes {
 		volume := pod.Spec.Volumes[i]
-		if volume.PersistentVolumeClaim == nil {
+		if _, ok := state.pvByVolume[volume.Name]; !ok {
```

Comment:
maybe directly iterate over `state.pvByVolume` instead of iterating over `pod.Spec.Volumes`